test/test_pycparserext.py

```diff
@@ -300,6 +300,38 @@ def test_array_ptr_decl_attribute():
     print(GnuCGenerator().visit(ast))
 
 
+def test_func_attribute_before_direct_declarator():
+    src = """
+    __attribute__((stdcall)) int func1(void);
+    int __attribute__((stdcall)) func2(void);
+    int func3(void) __attribute__((stdcall));
+    int (*fptr1)(int, int);
+    int (* __attribute__((stdcall)) fptr2)(int, int);
+    int (__attribute__((stdcall)) *fptr3)(int, int);
+    """
+    from pycparserext.ext_c_parser import GnuCParser
+    p = GnuCParser()
+    ast = p.parse(src)
+    ast.show()
+
+    from pycparserext.ext_c_generator import GnuCGenerator
+    src_regen = GnuCGenerator().visit(ast)
+    print(src_regen)
+
+    # _round_trip_matches trips at comparing the function
+    #   specifiers here, as they are lists
+    # (how was this supposed to work, again?)
```

Comment:
What do you mean here? If something has broken semantics, just fix it.

Reply:
I __think__ it has broken semantics, but I don't know how to fix it __yet__. It requires a bit more research into the codebase.